apps/elixir_ls_debugger/lib/debugger/server.ex

```diff
@@ -930,18 +919,18 @@ defmodule ElixirLS.Debugger.Server do
   end
 
   defp ensure_frame_id(state = %__MODULE__{}, pid, %Frame{} = frame) when is_pid(pid) do
-    unless Map.has_key?(state.paused_processes, pid) do
-      raise ArgumentError, message: "paused process #{inspect(pid)} not found"
-    end
+    paused_process = Map.fetch!(state.paused_processes, pid)
 
-    if Map.has_key?(state.paused_processes[pid].frames_inverse, frame) do
-      {state, state.paused_processes[pid].frames_inverse[frame]}
-    else
-      id = state.next_id
-      state = put_in(state.paused_processes[pid].frames[id], frame)
-      state = put_in(state.paused_processes[pid].frames_inverse[frame], id)
-      state = put_in(state.next_id, id + 1)
-      {state, id}
+    case paused_process.frames_to_frame_ids[frame] do
+      nil ->
```

Comment:
this pattern still does a map fetch and then a match; I don't see the advantage of a case if it's done like this.
My suggestion was:
```elixir
case paused_process.frames_to_frame_ids do 
  %{^frame => frame_id} -> 
    {state, frame_id}
   _ -> 
    id = state.next_id
    state = put_in(state.paused_processes[pid].frame_ids_to_frames[id], frame)
    state = put_in(state.paused_processes[pid].frames_to_frame_ids[frame], id)
    state = put_in(state.next_id, id + 1)
    {state, id}
end
```

Same with line #901 and #875